include/utility/dbus_utility.hpp

```diff
@@ -286,5 +286,40 @@ inline types::BiosAttributeCurrentValue
 
     return std::get<1>(l_attributeVal);
 }
+
+/**
+ * @brief API to check if Chassis is powered on.
+ *
+ * This API queries Phosphor Chassis State Manager to know whether
+ * Chassis is powered on.
+ *
+ * @return true if chassis is powered on, false otherwise
+ */
+inline bool isChassisPowerOn()
+{
+    auto powerState = dbusUtility::readDbusProperty(
+        "xyz.openbmc_project.State.Chassis",
+        "/xyz/openbmc_project/state/chassis0",
+        "xyz.openbmc_project.State.Chassis", "CurrentPowerState");
+
+    if (auto curPowerState = std::get_if<std::string>(&powerState))
+    {
+        if ("xyz.openbmc_project.State.Chassis.PowerState.On" == *curPowerState)
+        {
+            logging::logMessage("Chassis is in on state");
+            return true;
+        }
+        return false;
```

Comment:
Do we need this?

Reply:
Yes, we do. If `if ("xyz.openbmc_project.State.Chassis.PowerState.On" == *curPowerState)` fails, which means `curPowerState` is some other `std::string` like Transition On or Power Off, we **are** able to determine chassis state. So return false without logging a PEL.

Reply:
agree